json/rest/rest.go

```
@@ -0,0 +1,88 @@
+/*
+* Copyright(c) 2012-2021 Intel Corporation
+* SPDX-License-Identifier: BSD-3-Clause-Clear
+ */
+
+package rest
+
+import (
+	"fmt"
+	"net/http"
+
+	gin "github.com/gin-gonic/gin"
+)
+
+/** gin-web-framework based RESTful API **/
+
+func RestAPI() {
+	router := gin.Default()
+	router.GET("/", getHome)
+
+	// test of redirect
+	/*
+	router.GET("/log", func(c *gin.Context) {
+		c.Request.URL.Path = "/"
+		router.HandleContext(c)
+	})
+	*/
+
+	router.GET("/log", getLog)
+	router.POST("/request", postRequest)
+	router.GET("/request/", getRequest)
+	router.GET("/cacheList", getCacheList)
+	router.GET("/stats", getStats)
+	router.Run("localhost:9090")
+}
+
+func getHome(c *gin.Context) {
+	fmt.Fprintln(c.Writer, "Hello CAS")
+}
+
+func getRequest(c *gin.Context) {
+	var req Request
+	req.Read_req()
+	c.IndentedJSON(http.StatusOK, req)
+}
+
+func postRequest(c *gin.Context) {
+	var req Request
+	// req from client
+	if err := c.BindJSON(&req); err != nil {
+		return
+	}
+	c.IndentedJSON(http.StatusCreated, req)
+}
+
+/** Responds for HTTP request with retrived list of caches with their cores from CAS */
+
+func getCacheList(c *gin.Context) {
+	cache_list_pkg, _ := List_caches()
+	c.IndentedJSON(http.StatusCreated, cache_list_pkg)
+}
+
+/** Responds for HTTP request with retrived statistics and caching info from CAS */
+
+func getStats(c *gin.Context) {
+	var req Request
+	req.Read_req()
+
+	stats_pkg, _ := Get_stats(req.Get_stats)
```

Instead of ignoring error return it to the upper layer